config.py

```
@@ -41,8 +44,19 @@ def validate(cls):
         required = [
             "DISCORD_TOKEN",
             "OPENROUTER_API_KEY",
-            "OPENAI_API_KEY"
         ]
+
+        # Exigir OPENAI_API_KEY apenas quando o provedor de embeddings for explicitamente 'openai'
+        # ou quando estiver em modo 'auto' e a chave estiver presente (mantém compatibilidade).
+        provider = getattr(cls, "EMBEDDINGS_PROVIDER", "auto")
+        if provider == "openai":
+            required.append("OPENAI_API_KEY")
+        elif provider == "auto" and not getattr(cls, "OPENAI_API_KEY"):
```

Ruff (0.12.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)